package/libnetwork/src/libnetwork.c

```diff
@@ -165,6 +171,65 @@ static size_t network_upload_read(char *buf, size_t size, size_t n, void *data)
   return -1;
 }
 
+static void trim_crlf(char* in_buf, char* out_buf, size_t buf_size)
+{
+  strncpy(out_buf, in_buf, buf_size);
+  char* crlf = strstr(out_buf, "\r\n");
+  *crlf = '\0';
+}
+
+static size_t network_upload_write(char *buf, size_t size, size_t n, void *data)
+{
+  network_config_t *cfg = (network_config_t*) data;
```

Comment:
I liked the idea of passing in callback-specific data here (instead of the general config), but understand in that it's just the `->debug` that's getting used. If there's more things to add and pass in, should definitely have a standalone data structure.